tensorflow/core/kernels/transpose_op.cc

```diff
@@ -133,7 +133,16 @@ void TransposeOp::Compute(OpKernelContext* ctx) {
   OP_REQUIRES(ctx, TensorShapeUtils::IsVector(perm.shape()),
               errors::InvalidArgument("perm must be rank 1, got shape ",
                                       perm.shape().DebugString()));
-
+  // Validation for negative values inside `perm`.
+  auto perm_vector = perm.vec<int>();
```

Comment:
```suggestion
  auto perm_vector = perm.vec<Tperm>();
```

(similar to line 98, you just need to pass template argument)

Reply:
Thanks for it. I got a doubt here the TransposeOp::Compute() function don't have a `template` . Whether we can still pass `Tperm` here  or we need to have template also?

Reply:
It is a class member function and the class is templatized.